ProcessLib/GroundwaterFlow/GroundwaterFlowFEM.h

```diff
@@ -64,21 +77,31 @@ class LocalAssemblerData : public ProcessLib::LocalAssemblerInterface<GlobalMatr
         assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
     }
 
-    void assemble(double const /*t*/, std::vector<double> const& /*local_x*/) override
+    void assemble(double const /*t*/, std::vector<double> const& local_x) override
     {
         _localA.setZero();
         _localRhs.setZero();
 
         IntegrationMethod integration_method(_integration_order);
         unsigned const n_integration_points = integration_method.getNPoints();
 
-        for (std::size_t ip(0); ip < n_integration_points; ip++) {
+        for (std::size_t ip(0); ip < n_integration_points; ip++)
+        {
             auto const& sm = _shape_matrices[ip];
             auto const& wp = integration_method.getWeightedPoint(ip);
-
             auto const k = _process_data.hydraulic_conductivity(_element);
+
             _localA.noalias() += sm.dNdx.transpose() * k * sm.dNdx *
                                  sm.detJ * wp.getWeight();
+
+            // Darcy velocity only computed for output.
+            auto const darcy_velocity = (k * sm.dNdx *
```

Comment:
Better to move the computation of the velocity to another function, and also call this function in a standalone function outside of the global assembly function. If we deal with non-linear problems, we only need to compute the velocity after a linearization step is converged.